flink-connectors/flink-connector-kinesis/src/test/java/org/apache/flink/streaming/connectors/kinesis/proxy/KinesisProxyTest.java

```diff
@@ -86,4 +86,19 @@ protected AmazonKinesis createKinesisClient(Properties configProps) {
 		assertEquals(10000, clientConfiguration.getSocketTimeout());
 	}
 
+	@Test
+	public void testClientConfigOverride() {
+
+		Properties configProps = new Properties();
+		configProps.setProperty(AWSConfigConstants.AWS_REGION, "us-east-1");
+		configProps.setProperty(AWSUtil.AWS_CLIENT_CONFIG_PREFIX + "socketTimeout", "9999");
```

Comment:
Would this usage also subsume other commonly seen settings, such as setting the AWS crredentials?
Currently we have these explicitly defined keys in `AWSConfigConstants`, for which we might be able to get rid of (because we have to maintain the set of config keys ourselves, which is not nice and hard to meet with all the functionalities that KCL provides).

If so, we might also want tests + update our documents for those.

Reply:
The credentials construction code may be hard to replace declaratively. Also, the credentials are not a client config property, but a property of the Kinesis client.